File: sdks/flutter/installation-example/checkout_feedback.dart
import 'package:flutter/material.dart';
import 'package:likerts/likerts.dart';
import 'package:likerts/survey.dart';

/// The host owns eligibility, loading, cancellation and stable submission keys.
class CheckoutFeedback extends StatefulWidget {
  final Collection collection;
  final bool eligible, submitting;
  final ValueChanged<Map<String, dynamic>> onSubmit;
  final VoidCallback onDismiss;
  const CheckoutFeedback({super.key, required this.collection, required this.eligible,
    required this.submitting, required this.onSubmit, required this.onDismiss});
  @override State<CheckoutFeedback> createState() => _CheckoutFeedbackState();
}
class _CheckoutFeedbackState extends State<CheckoutFeedback> {
  bool visible = false;
  @override Widget build(BuildContext context) {
    if (!widget.eligible) return const SizedBox.shrink();
    return Column(children: [
      TextButton(onPressed: () => setState(() => visible = true), child: const Text('Give feedback')),
      if (visible) LikertsSurvey(collection: widget.collection, disabled: widget.submitting, onSubmit: widget.onSubmit),
      if (visible) TextButton(onPressed: () {setState(() => visible = false); widget.onDismiss();}, child: const Text('Dismiss')),
    ]);
  }
}

File: sdks/ios/InstallationExample/CheckoutFeedback.swift
import Likerts
import SwiftUI

// The host loads the collection, owns submission/retry state and controls
// eligibility. Dismissing the sheet must cancel the host's submission task.
public struct CheckoutFeedback: View {
    let collection: Collection
    let eligible: Bool
    let submitting: Bool
    let submit: ([String: Answer]) -> Void
    let cancel: () -> Void
    @State private var visible = false
    public init(collection: Collection, eligible: Bool, submitting: Bool,
                submit: @escaping ([String: Answer]) -> Void, cancel: @escaping () -> Void) {
        self.collection = collection; self.eligible = eligible; self.submitting = submitting
        self.submit = submit; self.cancel = cancel
    }
    public var body: some View {
        Button("Give feedback") { visible = true }
            .disabled(!eligible)
            .sheet(isPresented: $visible, onDismiss: cancel) {
                SurveyView(collection: collection, disabled: submitting, onSubmit: submit)
            }
    }
}
